### ISSUE TITLE
Hello, I trained libero with nearly identical configurations for 10,000 steps, but the final results show some slight deviations. May I ask how many steps your ultimately adopted model has?

### ISSUE THREAD

**Issue by aaaaaaapp** [starVLA/starVLA#56]
Hello, I trained libero with nearly identical configurations for 10,000 steps, but the final results show some slight deviations. May I ask how many steps your ultimately adopted model has?

**Comment by JinhuiYE** [comment]
Hi, I want to know more about which config you trained with?

**Comment by aaaaaaapp** [comment]
[run_libero_train.sh](https://github.com/user-attachments/files/24014293/run_libero_train.sh)

[starvla_cotrain_libero_1104.yaml](https://github.com/user-attachments/files/24014299/starvla_cotrain_libero_1104.yaml)

<img width="1438" height="648" alt="Image" src="https://github.com/user-attachments/assets/1d3471f3-17b8-4d52-aa27-12bc3390acb7" />

Mainly, there's a significant gap in the long-term performance, but everything else is fine. Besides, I noticed you only shared the libero model weights for qwen2.5vl (with 3000 steps), and not for qwen3vl.

**Comment by JinhuiYE** [comment]
libero model weights for qwen2.5vl (with 3000 steps), and not for qwen3vl.: hha it just due to too many ckpts and reach hf limitation

**Comment by JinhuiYE** [comment]
--trainer.gradient_accumulation_steps 2  might not work, 

plz set it in
starVLA/config/deepseeds/ds_config.yaml

**Comment by aaaaaaapp** [comment]
Thanks！In my previous training, 10,000 steps corresponded to over 15 epochs. Does this actually align with a gradient accumulation of 1 and a batch size (bs) of 8?

**Comment by JinhuiYE** [comment]
Yes, batch 16 per GPU, 8 GPU and 30k = 9.7 epoch

**Comment by aaaaaaapp** [comment]
I got it! Thanks!

**Comment by aaaaaaapp** [comment]
What's more, how many steps are appropriate for Qwen3VL? Additionally, why is the performance on Libero_10  not quite satisfactory?

**Comment by JinhuiYE** [comment]
your total batch size might too samll

**Comment by aaaaaaapp** [comment]
> your total batch size might too samll你的总批次数量也可能很差

Thanks

**Comment by SummerWXK** [comment]
I have successfully reproduced QwenPI based on Qwen3-VL-4B-Instruct, achieving 92.5% avg success rate (spatial: 95.8%, object: 99.4%, goal: 96.6%, long: 78.2%) after training 9.54 epoch similar to your setting, though libero-10's evaluatiom also still slightly below the reported results. The setting is 16 batch per GPU, 1w5 step, I want to know your results on Libero 10 using QwenPI based on Qwen3-VL-4B-Instruct.

**Comment by aaaaaaapp** [comment]
I didn't reproduce PI.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "starVLA/starVLA"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年12月9日(星期二) 下午5:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [starVLA/starVLA] Hello, I trained libero with nearly identical configurations for 10,000 steps, but the final results show some slight deviations. May I ask how many steps your ultimately adopted model has? (Issue #56)



SummerWXK left a comment (starVLA/starVLA#56)
 
I have successfully reproduced QwenPI based on Qwen3-VL-4B-Instruct, achieving 92.5% avg success rate (spatial: 95.8%, object: 99.4%, goal: 96.6%, long: 78.2%) after training 9.54 epoch similar to your setting, though libero-10's evaluatiom also still slightly below the reported results. The setting is 16 batch per GPU, 1w5 step, I want to know your results on Libero 10 using QwenPI based on Qwen3-VL-4B-Instruct.

**Comment by SummerWXK** [comment]
> I have successfully reproduced QwenPI based on Qwen3-VL-4B-Instruct, achieving 92.5% avg success rate (spatial: 95.8%, object: 99.4%, goal: 96.6%, long: 78.2%) after training 9.54 epoch similar to your setting, though libero-10's evaluatiom also still slightly below the reported results. The setting is 16 batch per GPU, 1w5 step, I want to know your results on Libero 10 using QwenPI based on Qwen3-VL-4B-Instruct.

@JinhuiYE Looking forward to your results！

**Comment by aaaaaaapp** [comment]
Do the results of **libero_10 (long)** obtained by reproducing other action head methods match the reported results?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "starVLA/starVLA"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年12月9日(星期二) 下午5:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [starVLA/starVLA] Hello, I trained libero with nearly identical configurations for 10,000 steps, but the final results show some slight deviations. May I ask how many steps your ultimately adopted model has? (Issue #56)



SummerWXK left a comment (starVLA/starVLA#56)
 
I have successfully reproduced QwenPI based on Qwen3-VL-4B-Instruct, achieving 92.5% avg success rate (spatial: 95.8%, object: 99.4%, goal: 96.6%, long: 78.2%) after training 9.54 epoch similar to your setting, though libero-10's evaluatiom also still slightly below the reported results. The setting is 16 batch per GPU, 1w5 step, I want to know your results on Libero 10 using QwenPI based on Qwen3-VL-4B-Instruct.

**Comment by SummerWXK** [comment]
> Do the results of **libero_10 (long)** obtained by reproducing other action head methods match the reported results?
> […](#)

Sorry, I have only reproduced Qwen3PI because it serves as the baseline for a piece of work I am currently undertaking.

**Comment by JinhuiYE** [comment]
Qwen3PI in WidowX:
step,PutCarrotOnPlateInScene,PutEggplantInBasketScene,PutSpoonOnTableClothInScene,StackGreenCubeOnYellowCubeBakedTexInScene,Average Across Tasks
10000,0.21875,0.6666666666666666,0.21875,0.05208333333333333,0.28906249999999994
20000,0.46875,0.5520833333333334,0.8229166666666667,0.28125,0.53125
30000,0.4791666666666667,0.9270833333333334,0.6666666666666667,0.22916666666666669,0.5755208333333334
40000,0.46875,0.8854166666666666,0.78125,0.30208333333333337,0.609375

@SummerWXK

**Comment by JinhuiYE** [comment]
Hi all,

Quick heads-up: I just fixed that annoying bug in `eval_libero.py`. So sorry about that!

It can reproduce results on libero_goal on my side. The patch is ready and will be released very soon."

<img width="1948" height="534" alt="Image" src="https://github.com/user-attachments/assets/5f220345-9f22-4aa2-95b7-4f024610f341" />

**Comment by JinhuiYE** [comment]
> > I have successfully reproduced QwenPI based on Qwen3-VL-4B-Instruct, achieving 92.5% avg success rate (spatial: 95.8%, object: 99.4%, goal: 96.6%, long: 78.2%) after training 9.54 epoch similar to your setting, though libero-10's evaluatiom also still slightly below the reported results. The setting is 16 batch per GPU, 1w5 step, I want to know your results on Libero 10 using QwenPI based on Qwen3-VL-4B-Instruct.
> 
> [@JinhuiYE](https://github.com/JinhuiYE) Looking forward to your results！


Here is our experimental results based on the code version as of December 10, 2025 (commit 4c987ea0d3f3750fda8b4499026957231fbf86b0).

Qwen3-VL-PI | 10k steps | 20k steps | 30k steps | 40k steps | 50k steps
-- | -- | -- | -- | -- | --
libero_goal | 0.752 | 0.834 | 0.958 | 0.948 | 0.974
libero_object | 0.85 | 0.886 | 0.996 | 0.982 | 0.998
libero_spatial | 0.896 | 0.952 | 0.988 | 0.986 | 0.984
libero_10 | 0.46 | 0.636 | 0.884 | 0.86 | 0.928




Qwen2.5-VL-PI | 10k steps | 20k steps | 30k steps | 40k steps | 50k steps
-- | -- | -- | -- | -- | --
libero_goal | 0.716 | 0.798 | 0.956 | 0.972 | 0.972
libero_object | 0.658 | 0.792 | 0.992 | 0.996 | 0.990
libero_spatial | 0.930 | 0.962 | 0.982 | 0.982 | 0.982
libero_10 | 0.542 | 0.516 | 0.884 | 0.894 | 0.926

**Comment by evelynly61** [comment]
> Hi all,
> 
> Quick heads-up: I just fixed that annoying bug in `eval_libero.py`. So sorry about that!
> 
> It can reproduce results on libero_goal on my side. The patch is ready and will be released very soon."
> 
> <img alt="Image" width="1948" height="534" src="https://private-user-images.githubusercontent.com/131999494/526273014-5f220345-9f22-4aa2-95b7-4f024610f341.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYzNzU3MjEsIm5iZiI6MTc2NjM3NTQyMSwicGF0aCI6Ii8xMzE5OTk0OTQvNTI2MjczMDE0LTVmMjIwMzQ1LTlmMjItNGFhMi05NWI3LTRmMDI0NjEwZjM0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyMlQwMzUwMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2Y1N2NlODRjZTExODRkMzE4YjNmMDdjM2FiODA4YTZkMTZkYmE5NGFjOTNjM2Q0OTQ0MWQyZjc0ZTJlZjFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gLH0b7Hp5Z-xLIULV3yCr6svu_OAHw4-yCcAAcQ9oKU">

Why does whether this step is passed in or not have such a significant impact?

**Comment by JinhuiYE** [comment]
LIBERO is sensitive to the consistency between training and eval.
